haas/api.py

```diff
@@ -258,34 +275,57 @@ def node_connect_network(node, nic, network):
 
     project = node.project
 
+    allocator = get_network_allocator()
+    if channel is None:
+        channel = allocator.get_default_channel(db)
+
```

Comment:
This isn't actually checking for an error --- None just indicates that it should use the default channel (one was not provided by the user).